xbmc/addons/interfaces/GUI/Window.cpp

```diff
@@ -280,6 +315,578 @@ bool Interface_GUIWindow::do_modal(void* kodiBase, void* handle)
 
   return true;
 }
+//@}
+
+/*!
+ * Window control functions
+ */
+//@{
+bool Interface_GUIWindow::set_focus_id(void* kodiBase, void* handle, int control_id)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUIAddonWindow* pAddonWindow = static_cast<CGUIAddonWindow*>(handle);
+  if (!addon || !pAddonWindow)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIWindow::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, pAddonWindow, addon ? addon->ID().c_str() : "unknown");
+    return false;
+  }
+
+  CGUIWindow *pWindow = dynamic_cast<CGUIWindow*>(g_windowManager.GetWindow(pAddonWindow->m_windowId));
```

Comment:
If pAddonWindow is of type <code>CGUIAddonWindow</code> which is guaranteed if you come across this code line, it is save that pAddonWindow is a <code>CGUIWindow</code>, because <code>CGUIAddonWindow</code> is derived from <code>CGUIMediaWindow</code> which is derived from <code>CGUIWindow</code>, right? You don't need pWindow here and everywhere in the code you introduced in this PR. You can call directly on pAddonWindow. ;-)